integration_tests/src/main/python/string_test.py

```diff
@@ -939,3 +906,33 @@ def test_format_number_float_value():
         lambda spark: unary_op_df(spark, data_gen).selectExpr('format_number(a, 5)').collect())))
     for cpu, gpu in zip(cpu_results, gpu_results):
         assert math.isclose(cpu, gpu, rel_tol=1e-7) or math.isclose(cpu, gpu, abs_tol=1.1e-5)
+
+# valid base range is [2, 36], to_base can be negative, out of range results nulls
+# TODO: test ansi mode
+@pytest.mark.parametrize('ansi_mode', [False], ids=idfn)
+def test_conv(ansi_mode):
+    gen = [
+        ("str_col", mk_str_gen(r'-?[0-9a-zA-Z]{1,15}')),
```

Comment:
We should also test implicit casts. IIRC this can be a number type in the user query

Reply:
I am concerned with the high probability that this is not going to do the right thing in a lot of cases? Like if I am going from base 2 to base anything. We have a super small probability that only '-', '0', and '1' end up in the output string. Even for base 35 if a z of Z shows up in any of the potentially 15 characters it is invalid.  I'm not sure the best way to fix this, but I don't like the current test.

Reply:
Added more cases:

```python
@pytest.mark.parametrize('from_base,pattern',
                         [
                             pytest.param(10, r'-?[0-9]{1,18}',       id='from_10'),
                             pytest.param(16, r'-?[0-9a-fA-F]{1,15}', id='from_16'),
                             pytest.param(36, r'-?[0-9a-zA-Z]{1,15}', id='from_36')
                         ])
@pytest.mark.parametrize('to_base', [2, 10, 16, -2, -10, -16], ids=idfn)
def test_conv_with_more_valid_values(from_base, to_base, pattern):
```